src/toil/wdl/wdltoil.py

```diff
@@ -1937,15 +1938,20 @@ def patched_run_invocation(*args: Any, **kwargs: Any) -> List[str]:
 
                     extra_flags: Set[str] = set()
                     accelerators_needed: Optional[List[AcceleratorRequirement]] = self.accelerators
+                    local_accelerators = get_individual_local_accelerators()
                     if accelerators_needed is not None:
                         for accelerator in accelerators_needed:
                             if accelerator['kind'] == 'gpu':
-                                if accelerator['brand'] == 'nvidia':
+                                # Grab detected GPUs
+                                local_gpus: List[Optional[str]] = [accel['brand'] for accel in local_accelerators if accel['kind'] == 'gpu'] or [None]
+                                # Tell singularity the GPU type
+                                gpu_brand = accelerator.get('brand') or local_gpus[0]
+                                if gpu_brand == 'nvidia':
                                     # Tell Singularity to expose nvidia GPUs
                                     extra_flags.add('--nv')
-                                elif accelerator['api'] == 'rocm':
+                                elif gpu_brand == 'amd':
                                     # Tell Singularity to expose ROCm GPUs
-                                    extra_flags.add('--nv')
+                                    extra_flags.add('--rocm')
```

Comment:
This won't handle the case where a task requests "2 GPUs" of unspecified types, and the machine has 2 GPUs, but they aren't the same brand, but the task is prepared for this and expects to e.g. poll what brands are available within the task and run two things in parallel that could take either brand of GPU. We'll always look at the first GPU and put the options needed to access it, even if we sort of already consumed it to satisfy the first required accelerator.

I think it probably isn't worth writing the code to support this case; only pedantic people with very unusual setups would try to make that work, and I'm not sure Singularity supports both these options at the same time anyway.

But we should add a note here about how that isn't going to work, so it doesn't look like we think this approach is technically correct.